source/server/http/admin.cc

```diff
@@ -314,6 +315,27 @@ Http::Code AdminImpl::handlerCerts(const std::string&, Buffer::Instance& respons
   return Http::Code::OK;
 }
 
+static constexpr char route_config_provider_manager_singleton_name[] = "route_config_provider_manager_singleton_name";
+
+Http::Code AdminImpl::handlerRoutes(const std::string&, Buffer::Instance& response) {
+  auto route_config_provider_manager =
+  server_.singletonManager().getTyped<Router::ServerRouteConfigProviderManager>(Router::route_config_provider_manager_singleton_name, [] {
+    // If there is no RouteConfigProviderManager in the server's Singleton::Manager, then
+    // there are no currently loaded Http Filter Stacks.
+    return nullptr;
+  });
+
+  if (route_config_provider_manager == nullptr) {
+    response.add("There are no dynamic route tables currently loaded.");
+    return Http::Code::OK;
```

Comment:
OK. BTW, I would strongly recommend saving the proto as seen by `rds_api_impl.cc` after https://github.com/lyft/envoy/pull/1441 instead of the JSON. This will make this forward compatible with the v2 work and it's super easy to print protos to a human readable format (`some_proto.DebugString()`).